app/controllers/auth_controller.rb

```diff
@@ -31,6 +31,15 @@ def fetch_return_to_path
   def do_user_auth(auth)
     sign_out
     user = User.from_oauth_hash(auth)
+    unless ENV["NO_WELCOME_EMAIL"]
+      send_welcome_mail(user)
+    end
     sign_in(user)
   end
+
+  def send_welcome_mail(user)
+    if (Time.current - user.created_at) < 2.seconds
+      WelcomeMailer.welcome_notification(user).deliver_later
```

Comment:
I think it's fine to keep it as-is. I do like having all emails triggered via dispatcher for grepping reason but I agree with you that this isn't client specific so not needed. Plus easy to move later if need be